dissect/target/plugins/os/unix/config.py

```diff
@@ -81,54 +105,216 @@ def parse_file(self, fh: io.TextIO) -> None:
+
+    parser = parser_type.create_parser(contents)
+
+    with entry.open() as fh:
+        open_file = io.TextIOWrapper(fh, encoding="utf-8")
+        parser.read_file(open_file)
+
+    return parser
+
+
+def _select_parser(entry: FilesystemEntry, hint: Optional[str] = None) -> ParserConfig:
+    if hint and (parser_type := CONFIG_MAP.get(hint)):
+        return parser_type
 
-    def __init__(self, target: Target, **kwargs):
+    extension = entry.path.rsplit(".", 1)[-1]
+
+    known_extension = CONFIG_MAP.get(extension, ParserConfig(Default))
+    return KNOWN_FILES.get(entry.name, known_extension)
+
+
+def create_entry(
+    fs, path: str, entry, hint: str, collapse: str, seperator: tuple[str], comment_prefixes: tuple[str]
+) -> FilesystemEntry:
+    if entry.is_file():
+        contents = ParserContents(collapse, seperator, comment_prefixes)
+        parser_items = parse_config(entry, hint, contents)
+        return ConfigurationEntry(fs, path, entry, parser_items=parser_items)
+    return entry
+
+
+class ConfigurationTreePlugin(Plugin):
+    __namespace__ = "config_tree"
+
+    def check_compatible(self) -> None:
+        # This should be able to be retrieved, regardless of OS
+        return None
+
+    @lru_cache(128)
+    def __call__(
+        self,
+        path: str = "/",
+        hint: Optional[str] = None,
+        collapse: Optional[Union[bool, set]] = None,
+        seperator: Optional[tuple[str]] = None,
+        comment_prefixes: Optional[tuple[str]] = None,
+    ):
+        """Create a configuration entry from a file, or a COnfigurationFilesystem from a directory.
+
+        If a directory is specified in ``path``, please provide the other arguments in the ``get`` call.
+
+        Args:
+            path: The path to either a directory or file
+            hint: What kind of parser it should use
+            collapse: Wether it should collapse all or only certain keys.
+            seperator: What seperator should be used for the parser.
+            comment_prefixes: What is specified as a comment.
+
+        """
+        target_path = self.target.fs.path(path)
+        file_path = target_path
+
+        while not file_path.exists():
+            file_path = file_path.parent
+
+        if file_path.is_file():
+            file_path = file_path.parent
+
+        fs = ConfigurationFilesystem(self.target, str(file_path))
+
+        output_path = target_path.relative_to(file_path)
+
+        if (path := str(output_path)) != ".":
+            return fs.get(path, file_path.get(), hint, collapse, seperator, comment_prefixes)
+        return fs
+
+    @internal
+    def get(
+        self,
+        path: Optional[str] = None,
+        hint: Optional[str] = None,
+        collapse: Optional[set] = None,
+        seperator: Optional[tuple[str]] = None,
+        comment_prefixes: Optional[tuple[str]] = None,
+    ):
+        return self.__call__(path or "/", hint, collapse, seperator, comment_prefixes)
```

Comment:
If you want a default for `path`, I'd set it in the input arguments. But I wonder if there should be a default.